activesupport/lib/active_support/core_ext/object/to_query.rb

```diff
@@ -72,18 +88,24 @@ class Hash
   #
   # The string pairs "key=value" that conform the query string
   # are sorted lexicographically in ascending order.
-  #
-  # This method is also aliased as +to_param+.
-  def to_query(namespace = nil)
+  def to_query(namespace = nil, &escape)
     query = filter_map do |key, value|
       unless (value.is_a?(Hash) || value.is_a?(Array)) && value.empty?
-        value.to_query(namespace ? "#{namespace}[#{key}]" : key)
+        value.to_query(namespace ? "#{namespace}[#{key}]" : key, &escape)
       end
     end
 
     query.sort! unless namespace.to_s.include?("[]")
     query.join("&")
   end
 
-  alias_method :to_param, :to_query
+  ##
+  # :method: to_param
+  # :call-seq: to_param()
+  #
+  # This method behaves like #to_query, but keys and values in the resulting
+  # query string are not escaped.
+  def to_param(namespace = nil)
```

Comment:
I've kept the `namespace` param here, just in case any existing code passed an argument to `to_param`, since it was an alias of `to_query`.  Though such code might break anyway, since `to_param` will no longer escape the square brackets that a `namespace` would add.  Perhaps it would be better to remove this param, or add a deprecation warning.